juju/juju
# feat(firewall): wire up firewaller to use ports domain

Wire up the firewaller api on the controller facade to be backed by
the DQLite ports domain.
 
This included substituting the DQLite watcher for the Mongo watcher.

This required a quick patch to the DQLite watcher to emit machine names
instead of machine uuids, since this data is consumed outside the
apiserver, so should be a natural key.

We also needed to wire up the OpenedPortRangesForMachine facade
endpoint. This required a bit of hacking (i.e. hardcoding SubnetCIDRs to
nil) to wire up. This was done since the application endpoints domain is
not yet wired up. Restoring this has been added to the risk register.

## Checklist


## QA steps

(Note: 4.0 tip is currently experiencing some difficulties with controllers (uniter bouncing). You may need to bootstrap a fw times to get a functioning controller)

Run
```
juju bootstrap aws aws
juju add-model m
juju deploy ubuntu
juju expose ubuntu
juju exec --unit ubuntu/0 open-port 1000/tcp
```

Check debug logs for:
```
machine-0: 11:51:28 INFO juju.provider.ec2 opened ports in security group juju-0a8403c4-10cc-4b80-89d3-77ae26bb4b3d-0: [1000/tcp from 0.0.0.0/0,::/0]
```

verify the port 1000 is open on the machine sec group
![image](https://github.com/user-attachments/assets/49f1a578-417d-433f-ac65-4e60dc229443)
